metrics/memory/analyze_heap_memory.py

```
@@ -0,0 +1,968 @@
+import argparse
+import bisect
+import glob
+import json
+import re
+import shlex
+import sys
+from collections import Counter, deque
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Tuple
```

Ruff (0.15.1)

[warning] Import from `collections.abc` instead: `Iterable`, `Sequence`

Import from `collections.abc`

(UP035)